compiler/fir/fir2ir/src/org/jetbrains/kotlin/fir/backend/ConversionUtils.kt

```diff
@@ -40,56 +43,83 @@ internal fun <T : IrElement> FirElement.convertWithOffsets(
 
 internal fun createErrorType(): IrErrorType = IrErrorTypeImpl(null, emptyList(), Variance.INVARIANT)
 
-fun FirTypeRef.toIrType(session: FirSession, declarationStorage: Fir2IrDeclarationStorage): IrType {
+fun FirTypeRef.toIrType(session: FirSession, declarationStorage: Fir2IrDeclarationStorage, irBuiltIns: IrBuiltIns): IrType {
     if (this !is FirResolvedTypeRef) {
         return createErrorType()
     }
-    return type.toIrType(session, declarationStorage)
+    return type.toIrType(session, declarationStorage, irBuiltIns)
 }
 
-fun ConeKotlinType.toIrType(session: FirSession, declarationStorage: Fir2IrDeclarationStorage, definitelyNotNull: Boolean = false): IrType {
+fun ConeKotlinType.toIrType(
+    session: FirSession,
+    declarationStorage: Fir2IrDeclarationStorage,
+    irBuiltIns: IrBuiltIns,
+    definitelyNotNull: Boolean = false
+): IrType {
     return when (this) {
         is ConeKotlinErrorType -> createErrorType()
         is ConeLookupTagBasedType -> {
-            val firSymbol = this.lookupTag.toSymbol(session) ?: return createErrorType()
-            val irSymbol = firSymbol.toIrSymbol(session, declarationStorage)
+            var irSymbol = getPrimitiveArrayType(this.classId, irBuiltIns)
```

Comment:
Can we keep this as a `val`? Maybe something like:
```kt
val irSymbol = getPrimitiveArrayType(this.classId, irBuiltIns) ?: run {
  val firSymbol = this.lookupTag.toSymbol(session) ?: return createErrorType()
  firSymbol.toIrSymbol(session, declarationStorage)
}

// Or...
val irSymbol = getPrimitiveArrayType(this.classId, irBuiltIns) ?: this.lookupTag.toSymbol(session)?.toIrSymbol(session, declarationStorage) ?: return createErrorType()
```

Reply:
done